.circleci/config.yml

```diff
@@ -59,220 +63,103 @@ jobs:
     steps:
       - attach_workspace:
           at: /home/circleci/
       - run:
           command: |
-            RUN_JOB=`cat .ci-temp/run_job`
-            echo "RUN_JOB="$RUN_JOB
-            if [[ $RUN_JOB == 1 ]]; then
-              echo "./.ci/pitest.sh pitest-blocks"   >> commands.txt
-              echo "./.ci/pitest.sh pitest-sizes"    >> commands.txt
-              echo "./.ci/pitest.sh pitest-modifier" >> commands.txt
-              echo "./.ci/pitest.sh pitest-regexp"   >> commands.txt
-              CMD="$(circleci tests split commands.txt)"
-              echo "Command: $CMD"
-              eval $CMD
-            else
-              echo "build is skipped ..."
-            fi
-      - store_artifacts:
-          path: ./target/pit-reports
+            sudo apt install curl
+            curl -s get.sdkman.io | bash
+            source "$HOME/.sdkman/bin/sdkman-init.sh"
+            sdk install groovy
+            groovy -version
+            ./.ci/no-exception-test.sh no-exception-hadoop-apache-groovy-scouter-javadoc
 
-  pitest7:
-    machine: true
-    parallelism: 4
+  no-exception-only-javadoc:
+    machine:
+      image: ubuntu-2004:202010-01
+    parallelism: 1
     steps:
       - attach_workspace:
           at: /home/circleci/
       - run:
           command: |
-            RUN_JOB=`cat .ci-temp/run_job`
-            echo "RUN_JOB="$RUN_JOB
-            if [[ $RUN_JOB == 1 ]]; then
-              echo "./.ci/pitest.sh pitest-gui"   >> commands.txt
-              # echo "./.ci/pitest.sh pitest-meta"  >> commands.txt
-              CMD="$(circleci tests split commands.txt)"
-              echo "Command: $CMD"
-              eval $CMD
-            else
-              echo "build is skipped ..."
-            fi
-      - store_artifacts:
-          path: ./target/pit-reports
+            sudo apt install curl
+            curl -s get.sdkman.io | bash
+            source "$HOME/.sdkman/bin/sdkman-init.sh"
+            sdk install groovy
+            groovy -version
+            ./.ci/no-exception-test.sh no-exception-only-javadoc
 
-  pitest8:
-    machine: true
-    parallelism: 4
+  no-error-xwiki:
+    machine:
+      image: ubuntu-2004:202010-01
+    parallelism: 1
     steps:
       - attach_workspace:
           at: /home/circleci/
       - run:
           command: |
-            RUN_JOB=`cat .ci-temp/run_job`
-            echo "RUN_JOB="$RUN_JOB
-            if [[ $RUN_JOB == 1 ]]; then
-              echo "./.ci/pitest.sh pitest-common-2"   >> commands.txt
-              CMD="$(circleci tests split commands.txt)"
-              echo "Command: $CMD"
-              eval $CMD
-            else
-              echo "build is skipped ..."
-            fi
-      - store_artifacts:
-          path: ./target/pit-reports
+            sudo apt install curl
+            curl -s get.sdkman.io | bash
+            source "$HOME/.sdkman/bin/sdkman-init.sh"
+            sdk install groovy
+            groovy -version
```

Comment:
I think it is easier to just take image that has groovy pre-installed

Reply:
But, the pre-built images like `circleci/buildpack-deps:groovy` or others don't work. It is showing the same error of `groovy: command not found` . Tried to install groovy on a docker image of `circleci/openjdk:stretch`, by `apt-get install groovy` but throws `404 not found` error on two packages. Seems this is the only way out currently.

Reply:
there is no need to use circleci images.
You can take any good official groovy image https://hub.docker.com/_/groovy

Reply:
Used my own version of [groovy:jdk11](https://hub.docker.com/layers/groovy/library/groovy/jdk11/images/sha256-ffec94ae0c0eb1b2a906d92d81d0d4002f0a35e2df99da7d23a76c3053c392f3?context=explore), diff being added git and maven in container.

Reply:
@strkkk Please complete the review

Reply:
What is the difference between your image and official groovy jdk11 image?

If there is none, it is better to use official one to avoid issues (e.g. if you delete image for some reason), if there are some difference (some files added, some setup, etc.) - please share dockerfile - add it to contribution repo with good comments

Reply:
> What is the difference between your image and official groovy jdk11 image?

Only difference is that we have include maven and git in the image. We are using the official image from https://archive.apache.org/dist/groovy/

Please see https://github.com/checkstyle/contribution/pull/562/

Reply:
ok, that is fine.
the only thing that bugs me is that best place for this image is checkstyle docker repo.
@romani what do you think?

Reply:
> he only thing that bugs me is that best place for this image is checkstyle docker repo.

True, please use the dockerfile in contribution PR https://github.com/checkstyle/contribution/pull/562 .

Reply:
docker image uploaded to checkstyle dockerhub